3p-ecosystem/src/main/java/com/google/homesampleapp/screens/inspect/InspectFragment.kt

```diff
@@ -85,108 +116,156 @@ class InspectFragment : Fragment() {
 
-  // -----------------------------------------------------------------------------------------------
-  // UI update functions
+    binding.topAppBar.title = selectedDevice?.device?.name
 
-  private fun updateDeviceInfo() {
-    Timber.d("updateDeviceIfo")
-    if (selectedDeviceViewModel.selectedDeviceIdLiveData.value == -1L) {
-      // Device was just removed, nothing to do. We'll move to HomeFragment.
-      return
+    LifecycleResumeEffect {
+      Timber.d("LifecycleResumeEffect: selectedDeviceId [$selectedDeviceId]")
+      viewModel.inspectDevice(selectedDeviceViewModel.selectedDeviceLiveData.value!!.device.deviceId)
+      onPauseOrDispose {
+        // do any needed clean up here
+        Timber.d("LifecycleResumeEffect:onPauseOrDispose")
+      }
     }
-    val deviceUiModel = selectedDeviceViewModel.selectedDeviceLiveData.value
-    Timber.d(
-        "updateDeviceIfo with device [${deviceUiModel?.device?.deviceId}] [${deviceUiModel?.device?.name}]")
-
-    binding.topAppBar.title = deviceUiModel?.device?.name
-    // Fetch device introspection info, and then live data will be updated, which is observed
-    // by the fragment.
-    viewModel.inspectDevice(deviceUiModel?.device?.deviceId!!)
+
+    InspectScreen(selectedDeviceId, instrospectionInfo)
   }
 
-  private fun updateIntrospectionInfo(deviceMatterInfoList: List<DeviceMatterInfo>) {
-    val linearLayout = binding.inspectInfoLayout
-    linearLayout.removeAllViews()
-    if (deviceMatterInfoList.isEmpty()) {
-      addTextView(
-          "Oops... We could not retrieve any information from the Descriptor Cluster. " +
-              "This is probably because the device just recently turned \"offline\".",
-          com.google.android.material.R.style.TextAppearance_Material3_BodyMedium)
+  @Composable
+  private fun InspectScreen(
+    selectedDeviceId: Long?,
+    deviceMatterInfoList: List<DeviceMatterInfo>?
+  ) {
+    if (selectedDeviceId == -1L) {
+      // Device was just removed, nothing to do. We'll move to HomeFragment.
       return
     }
-    // Add the Descriptor Cluster Title
-    addTextView(
-        "Descriptor Cluster",
-        com.google.android.material.R.style.TextAppearance_Material3_TitleLarge)
-    // For each endpoint
-    for (deviceMatterInfo in deviceMatterInfoList) {
-      // Endpoint ID
-      addTextView(
-          "Endpoint ${deviceMatterInfo.endpoint}",
-          com.google.android.material.R.style.TextAppearance_Material3_TitleMedium)
-      // Device Types
-      addTextView(
-          "Device Types", com.google.android.material.R.style.TextAppearance_Material3_TitleSmall)
-      for (deviceType in deviceMatterInfo.types) {
-        val hex = String.format("0x%04X", deviceType)
-        val typeString = MatterConstants.DeviceTypesMap.getOrDefault(deviceType, "Unknown")
-        addTextView(
-            "[${hex}] ${typeString}",
-            com.google.android.material.R.style.TextAppearance_Material3_BodyMedium)
+    Column {
```

Comment:
Yeah, but then it triggers this runtime exception:

> java.lang.IllegalStateException: Vertically scrollable component was measured with an infinity maximum height constraints, which is disallowed. One of the common reasons is nesting layouts like LazyColumn and Column(Modifier.verticalScroll()). If you want to add a header before the list of items please add a header as a separate item() before the main items() inside the LazyColumn scope. There are could be other reasons for this to happen: your ComposeView was added into a LinearLayout with some weight, you applied Modifier.wrapContentSize(unbounded = true) or wrote a custom layout. Please try to remove the source of infinite constraints in the hierarchy above the scrolling container.

I've tried different things, but then gave up. 
Unless you have a quick fix, I think it will be easier to fix these things once I migrate all the header stuff (top appbar, etc) + navigation to compose (will be the last step, after I finish migrating all the views).